tensilelite/Tensile/KernelWriterAssembly.py

```diff
@@ -8323,279 +8417,290 @@ def globalWriteElements(self, kernel, tPA, tPB, vectorWidths, elements,
-          ########################################
-
-          ss = StoreState(self, kernel, gwvw, edge, beta, atomic, elements[edgeI])
-
-          # how many vgprs are needed for zero elements
-          # 2 for addressC in vgpr for addition - already checked out
-          # 2 for coord0,1 of thread - already checked out
-          # 2 for tmp - already checked out
-
-          # 5 = how many vgprs are needed per element (flat)
-          #  - 2 for addr
-          #  - 3 for GLOBAL_OFFSET_C calculation (can overlap below, therefore max)
-          #  - if beta gwvw*rpe for new value
-          #  - if atomic 2*rpe for old and cmp values
-
-          #print self.vgprPool.state()
-          # Use VGPR up to next occupancy threshold:
-          maxVgprs = self.getMaxRegsForOccupancy(kernel["NumThreads"], self.vgprPool.size(), \
-                                                self.getLdsSize(kernel), self.agprPool.size(), self.states.doubleVgpr)
-          if self.states.serializedStore: # get aggressive when serializedStore is on; not necessarily exclusive to this parameter
-            _growPool(self.vgprPool, self.vgprPool.size()-self.vgprPool.available(), maxVgprs, 1, \
-              "grow-pool up to next occupancy for GlobalWrite")
-          # Get numVgprAvailable
-          numVgprAvailable = self.vgprPool.availableBlock(ss.numVgprsPerElement, ss.align)
-
-          # Grow the register pool if needed - we need enough regs for at least one element
-          # Unfortunate since this means the write logic is setting the VGPR requirement
-          # for the entire kernel but at least we have a functional kernel.
-          # Before growing the pool, see if we can shrink the write vector width instead?
-          # TODO : the vgprSerial is needed for-ever and if we grow here will split the
-          # range of the tmps.  Maybe want to move vgprSerial to first vgpr?
-
-          # TODO: Minimum elems for StoreRemap
-          # TODO: Which of DataType or DestDataType is in a better sense? 0114: Check Using DestDataType + HSS
-          minElements = 2 if (kernel["ProblemType"]["DataType"].isHalf() or kernel["ProblemType"]["DataType"].isBFloat16()) else 1
-          minNeeded = minElements * ss.numVgprsPerElement
-          shrinkDb = 0
-          if shrinkDb:
-            print("numVgprAvailable=", numVgprAvailable, "minElements=", minElements, "minNeeded=", minNeeded)
-          if numVgprAvailable < minNeeded:
-            gwvwOrig = gwvw
-            currentOccupancy = self.getOccupancy(kernel["NumThreads"], self.getLdsSize(kernel), \
-                self.vgprPool.size(), self.agprPool.size(), self.states.doubleVgpr)
-            futureOccupancy = self.getOccupancy(kernel["NumThreads"], self.getLdsSize(kernel), \
-                self.vgprPool.size() - numVgprAvailable + minNeeded, self.agprPool.size(), self.states.doubleVgpr)
+          edge_mode_pos = 0
+          for idx2 in range(len(biasDims)):
```

Comment:
add new layer idx2 for generating global write batch for the different bias dims